### ISSUE TITLE
equalTo() fails to find null values on an indexed field if using Case.INSENSITIVE

### ISSUE THREAD

**Issue by morris86** [realm/realm-java#5299]
It appears that Realm versions from 3.1.3 onward have a bug where using equalTo() on a query no longer finds indexed fields which have null values, _but only when the `Case.INSENSITIVE` flag is being used_.  Using `Case.SENSITIVE` (the default) still works.

In the [Realm Java changelog for 3.1.3](https://github.com/realm/realm-java/blob/master/CHANGELOG.md#313-2017-04-20) we can see the following under enhancements:
> Upgraded to Realm Core 2.6.1.

We can see in the [Realm Core changelog for 2.6.0](https://github.com/realm/realm-core/blob/master/CHANGELOG.md#260-release-notes) the following, which looks like it could be the culprit:
> Made case insensitive queries use the new index based case insensitive search. PR [#2486](https://github.com/realm/realm-core/pull/2486)

#### Expected Results

> I would expect to be able to find records whose values for the field we're searching on are `null`

#### Actual Results

> No records are found

#### Code Sample

`RealmTest.class` is a subclass of `RealmObject` and defines `test_field` like so:
```java
    @Index
    private String test_field;
```

Now the query:
```java
    Realm realm = Realm.getDefaultInstance();

    String searchString = "test";
    realm.where(RealmTest.class)
        .equalTo("test_field", searchString, Case.INSENSITIVE)
        .findAll();
// 3.1.0 behaviour: returns all results where test_field has a value of "test" ✅
// 3.1.3 behaviour: returns all results where test_field has a value of "test" ✅


    searchString = null;
    realm.where(RealmTest.class)
        .equalTo("test_field", searchString, Case.INSENSITIVE)
        .findAll();
// 3.1.0 behaviour: returns all results where test_field has a value of null ✅
// 3.1.3 behaviour: does NOT return results where test_field has a value of null ❌
```

Again, this bug is only seen when `Case.INSENSITIVE` is used, which makes me suspicious that [Realm-Core PR #2486](https://github.com/realm/realm-core/pull/2486) *might* be the cause of this issue, but I'm not sure.

#### Version of Realm and tooling
Realm version(s): >= 3.1.3

Realm sync feature enabled: no

Android Studio version: 2.3.3

Which Android version and device: 8.0, Pixel.

**Comment by Zhuinden** [comment]
But Core 2.8.6 fixed some bugs regarding index string case insensitive query https://github.com/realm/realm-core/blob/master/CHANGELOG.md#286-release-notes

which means this might have been fixed in 3.5.0+ (as it is core 2.8.6), any change this is true?

**Comment by morris86** [comment]
Unfortunately not.. the problem still exists in 3.7.2.

**Comment by Zhuinden** [comment]
Hmm. Technically I think `isNull().or().equalTo()` would work, but I'll open an issue at the Core.

**Comment by ironage** [comment]
@morris86 thanks for reporting this!
I could reproduce this in core and made a fix in https://github.com/realm/realm-core/pull/2871
As @Zhuinden says, if this is a blocker for you now, you could make a special case and create a query `isNull().or().equalTo()` until the fix is released.

**Comment by morris86** [comment]
@ironage brilliant, thanks for the super fast turnaround on this one!

**Comment by cmelchior** [comment]
Fixed in 4.0.0-RC1. Closing
